e2e/api/steps.js

```diff
@@ -254,13 +255,14 @@ const assertValidData = async (data, pageId) => {
   assert.deepEqual(responseBody.data, caseData);
 };
 
-const assertCallbackError = async (pageId, eventData, expectedErrorMessage) => {
+const assertCallbackError = async (pageId, eventData, expectedErrorMessage, responseBodyMessage = 'Unable to proceed because there are one or more callback Errors or Warnings' ) => {
   const response = await apiRequest.validatePage(eventName, pageId, {...caseData, ...eventData}, 422);
   const responseBody = await response.json();
-
   assert.equal(response.status, 422);
-  assert.equal(responseBody.message, 'Unable to proceed because there are one or more callback Errors or Warnings');
-  assert.equal(responseBody.callbackErrors[0], expectedErrorMessage);
+  assert.equal(responseBody.message, responseBodyMessage);
+  if(responseBody.callbackErrors != null){
```

Comment:
is it null when we pass value that does not match regex?

Reply:
@ckalista I think it is much better for it to be Text+regex, also when using Number, field interpolation doesn't work

@pd-dudkiewicz This was done so that it could assert the error correctly, it is null because instead of giving a callback error its just returning "Case data validation failed" - so to answer your question even if we pass a value that does match the regex assuming there's no other callback errors it should also be null

Reply:
callback errors is only populated for callbacks, aka validation in our code. This regex validation is not in the callback and therefor this block is empty. Works with existing tests, we can change if the need requires